harness/determined/pytorch/_pytorch_context.py

```diff
@@ -440,6 +440,7 @@ def step_optimizer(
         optimizer: torch.optim.Optimizer,  # type: ignore
         clip_grads: Optional[Callable[[Iterator], None]] = None,
         auto_zero_grads: bool = True,
+        on_before_zero_grad: Optional[Callable[[torch.optim.Optimizer], None]] = None,
```

Comment:
this wouldn't be a breaking change would it?

Reply:
hmm why do we have the auto_zero_grads param then? 

also separately isn't something wrong with this check? 
```
        check.true(
            auto_zero_grads or self.hvd_config.aggregation_frequency > 1,
            "if optimizations.aggregation_frequency is larger than 1, "
            "you can only set auto_zero_grads to be true. ",
        )
```

Reply:
blocking: Actually, I take the word back. We shouldn't add this callback. The Pytorch interface is meant to let users call anything they want directly in `train_batch` rather than using callbacks. You should remove this callback.

`auto_zero_grads` is just for users who want to call zero grading themselves.

>  also separately isn't something wrong with this check?

Do you mean the check itself is wrong or the message?

Reply:
I removed it in the last review, there are no `auto_zero_grads` callbacks.


> Do you mean the check itself is wrong or the message?

￼the message? I don't know enough about hvd_config.aggregation_frequency to know for sure.

Reply:
TODO remove the callback from the argument list

Reply:
I included the fix for the aggregation frequency here.